source/extensions/extensions_build_config.bzl

```diff
@@ -243,15 +243,13 @@ EXTENSIONS = {
     #
 
     "envoy.rate_limit_descriptors.expr":                "//source/extensions/rate_limit_descriptors/expr:config",
-    
+
     #
     # IO socket
     #
 
     "envoy.io_socket.user_space":                       "//source/extensions/io_socket/user_space:config",
 }
 
-# These can be changed to ["//visibility:public"], for  downstream builds which
-# need to directly reference Envoy extensions.
-EXTENSION_CONFIG_VISIBILITY = ["//:extension_config"]
-EXTENSION_PACKAGE_VISIBILITY = ["//:extension_library"]
+EXTENSION_CONFIG_VISIBILITY = ["//visibility:public"]
```

Comment:
not sure if this change was necessary...

Reply:
so from playing with this further the `//source` refs are required and afaict this needs to be public for that to work

Reply:
Can you explain further? I think the objective of limited visibility here is to restrict how core can depend on extensions, which is something we want I think.

Reply:
ive added `//tool/extensions` into the `//:extension_config` package config 

i think this is the correct way to allow the tool access to the code

im learning bazel on the hoof, so apologies if my approach can be a bit hacky